app/controllers/events_controller.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+class EventsController < ApplicationController
+  def index
+    @events = Event.all
+    @upcoming = Event.upcoming
+    @past = Event.past
+  end
+
+  def new
+    @event = current_user.events.build
+  end
+
+  def create
+    @event = Event.new(event_params)
+    if @event.save
+      flash[:success] = 'Your account was created successfully'
+      redirect_to @event
+    else
+      flash[:danger] = 'User was not created'
+      render 'new'
+    end
+  end
+
+  def show
+    @event = Event.find(params[:id])
+    @creator = User.find(@event.user_id)
+  end
```

Comment:
- [ ] You should use your created associations to list the created events by the current user, you can also use the current_user helper

- [ ] we should also have the upcoming and past events instantiated in [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png)